src/curl_multi.cpp

```diff
@@ -271,35 +269,36 @@ int S3fsMultiCurl::MultiRead()
 
         if(isPostpone){
             clist_req.erase(iter);
-            clist_req.push_back(s3fscurl);    // Re-evaluate at the end
+            clist_req.push_back(std::move(s3fscurl));    // Re-evaluate at the end
             iter = clist_req.begin();
         }else{
             if(!isRetry || 0 != result){
                 // If an EIO error has already occurred, it will be terminated
                 // immediately even if retry processing is required. 
                 s3fscurl->DestroyCurlHandle();
-                delete s3fscurl;
             }else{
-                S3fsCurl* retrycurl = NULL;
+                std::unique_ptr<S3fsCurl> retrycurl;
 
                 // Reset offset
                 if(isNeedResetOffset){
-                    S3fsCurl::ResetOffset(s3fscurl);
+                    S3fsCurl::ResetOffset(s3fscurl.get());
                 }
 
                 // For retry
+                S3fsCurl* retry_ptr = nullptr;
                 if(RetryCallback){
-                    retrycurl = RetryCallback(s3fscurl);
+                    retry_ptr = RetryCallback(s3fscurl.get());
+                    retrycurl.reset(retry_ptr);
                     if(NULL != retrycurl){
```

Comment:
Shouldn't `if(NULL != retrycurl){` be changed to `if(retrycurl.get()){`?

Reply:
Changed to `retry_ptr` (same thing).